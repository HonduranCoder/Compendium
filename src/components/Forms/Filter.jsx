import React from 'react';
//Dropdown
export default function Filter({ types, filterChange, selectedType }) {
  return (
    <div>
      <select
        aria-label="dropdown"
        value={selectedType}
        onChange={(e) => filterChange(e.target.value)}
      >
        <option value="all">All</option>
        {types.map((type) => (
          <option key={type} value={type}>
            {type}
          </option>
        ))}
      </select>
    </div>
  );
}
