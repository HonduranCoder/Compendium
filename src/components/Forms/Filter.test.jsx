import { screen, render } from '@testing-library/react';
import Filter from './Filter.jsx';

it('test filter', () => {
  const types = ['Fire', 'Bug', 'Water'];
  render(<Filter types={types} />);
  const dropdown = screen.getByLabelText('dropdown');
  expect(dropdown).toMatchSnapshot();
});
