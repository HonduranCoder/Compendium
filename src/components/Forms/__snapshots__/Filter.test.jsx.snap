// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test filter 1`] = `
<select
  aria-label="dropdown"
>
  <option
    value="all"
  >
    All
  </option>
  <option
    value="Fire"
  >
    Fire
  </option>
  <option
    value="Bug"
  >
    Bug
  </option>
  <option
    value="Water"
  >
    Water
  </option>
</select>
`;
